pkg/controllers/provisioning/scheduling/topology_test.go

```diff
@@ -400,6 +400,26 @@ var _ = Describe("Topology", func() {
 			}
 			Expect(nodeNames).To(HaveLen(1))
 		})
+		It("should respect minDomains constraints", func() {
+			var minDomains int32 = 4
+			provisioner.Spec.Requirements = []v1.NodeSelectorRequirement{
+				{Key: v1.LabelTopologyZone, Operator: v1.NodeSelectorOpIn, Values: []string{"test-zone-1", "test-zone-2", "test-zone-3"}}}
+			topology := []v1.TopologySpreadConstraint{{
+				TopologyKey:       v1.LabelTopologyZone,
+				WhenUnsatisfiable: v1.DoNotSchedule,
+				LabelSelector:     &metav1.LabelSelector{MatchLabels: labels},
+				MaxSkew:           1,
+				MinDomains:        &minDomains,
+			}}
+			ExpectApplied(ctx, env.Client, provisioner)
+			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov,
+				test.UnschedulablePod(test.PodOptions{ObjectMeta: metav1.ObjectMeta{Labels: labels}, TopologySpreadConstraints: topology}),
+				test.UnschedulablePod(test.PodOptions{ObjectMeta: metav1.ObjectMeta{Labels: labels}, TopologySpreadConstraints: topology}),
+				test.UnschedulablePod(test.PodOptions{ObjectMeta: metav1.ObjectMeta{Labels: labels}, TopologySpreadConstraints: topology}),
+				test.UnschedulablePod(test.PodOptions{ObjectMeta: metav1.ObjectMeta{Labels: labels}, TopologySpreadConstraints: topology}),
+			)
+			ExpectSkew(ctx, env.Client, "default", &topology[0]).To(ConsistOf(1, 1, 2))
```

Comment:
@tzneal this is probably not the exact scenario you described [here](https://github.com/aws/karpenter-core/pull/330#discussion_r1190317848), but regardless I would expect if we implement support for `minDomains` properly we will not get a "1,1,2" SKU here, but rather some variety of "1,1,1" and then some error about the 4th pod not having an available domain for scheduling (so perhaps the assertion in L419 above fails)

In any event this test is passing, which unsurprisingly reveals that the current implementation isn't there yet, but is this a valid testing surface are to do basic functional testing of `minDomains`?

thx!

Reply:
Yeah, I don't see anything wrong with this.  I think you should get a skew of 1, 1 and 1 and an unschedulable pod.

Reply:
Cool, so the fact that this test is passing suggests the implementation isn't doing anything. In addition to the half-baked `nextDomainTopologySpread`, what are we overlooking in the topology business logic that isn't doing the right thing?

Reply:
Corrected the business logic so things are actually happening!